ThunkLibs/libvulkan/Host.cpp

```diff
@@ -56,7 +55,25 @@ static VkResult FEXFN_IMPL(vkCreateShaderModule)(VkDevice a_0, const VkShaderMod
   return LDR_PTR(vkCreateShaderModule)(a_0, a_1, nullptr, a_3);
 }
 
+static VkBool32 DummyVkDebugReportCallback(VkDebugReportFlagsEXT, VkDebugReportObjectTypeEXT, uint64_t, size_t,
+                                           int32_t, const char*, const char*, void*) {
+  return VK_FALSE;
+}
+
 static VkResult FEXFN_IMPL(vkCreateInstance)(const VkInstanceCreateInfo* a_0, const VkAllocationCallbacks* a_1, VkInstance* a_2) {
+  VkDebugReportCallbackCreateInfoEXT stub_debug_report;
+  const VkBaseInStructure* vk_struct = reinterpret_cast<const VkBaseInStructure*>(a_0);
+  for (; vk_struct->pNext; vk_struct = vk_struct->pNext) {
+    // Override guest callbacks used for VK_EXT_debug_report
+    if (*reinterpret_cast<const VkStructureType*>(vk_struct->pNext) == VK_STRUCTURE_TYPE_DEBUG_REPORT_CALLBACK_CREATE_INFO_EXT) {
+      memcpy(&stub_debug_report, a_0->pNext, sizeof(stub_debug_report));
+      stub_debug_report.pfnCallback = DummyVkDebugReportCallback;
+      // Overwrite the pNext pointer, ignoring its const-qualifier
+      memcpy(const_cast<void*>(a_0->pNext), &stub_debug_report, sizeof(a_0->pNext));
```

Comment:
whew, spooky. Easy to catch though.
I guess this will just be temporary while other work is happening to get these debug callbacks working correctly?

Reply:
Yes, we can remove this code once support for generic guest-callbacks is added, which is very soon.

Reply:
Sounds good to me